integration-tests/bats/sql-checkout.bats

```diff
@@ -584,6 +584,19 @@ SQL
   [[ "$output" =~ "0" ]] || false
 }
 
+@test "sql-checkout: checkout new branch should use revision database name to access DbData on current database" {
+    skiponwindows "Need to install expect and make this script work on windows."
+    mkdir doltsql
+    cd doltsql
+
+    run expect $BATS_TEST_DIRNAME/sql-checkout.expect
```

Comment:
I'm wondering if there's a good way to avoid using `expect` here... It seems like it might actually be less lines (and one less file) if the commands were just here in the BATS `@test` method. (and maybe this test would also run on windows that way and not need a skip) It also lets us check the output of the tests more deeply, which we don't do with `expect`.

I know you mentioned having some problems getting all the commands to run in the same session with `dolt sql`, which makes sense. You'd have to shove them into the same `dolt sql -q '...'` line to get them to use the same session, which doesn't sound very usable. :-) 

Alternatively... how about testing this one with an engine ScriptTest as part of the Dolt ScriptTests? It seems like that would ensure that you are using the same session for the commands.